clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -491,6 +491,9 @@ public boolean rejoinNeededOrPending() {
      * @return true iff the operation completed within the timeout
      */
     public boolean refreshCommittedOffsetsIfNeeded(Timer timer) {
+        // no committed offset should be fetched for the default group id
+        if (groupId == null) return true;
```

Comment:
Perhaps now we don't need this? We could add an assertion in the constructor that the groupId is not null.